src/roman.rs

```diff
@@ -137,24 +80,42 @@ impl Roman {
         buf
     }
 
+    /// Returns a [`RomanFormatter`] which lazily formats a `self` value as a lowercase or uppercase Roman numeral depending of `style`.
+    ///
+    /// ## Examples
+    ///
+    /// ```
+    /// use xvii::{Roman, Style};
+    ///
+    /// let value = Roman::new(12).unwrap();
+    /// assert_eq!(format!("{}", value.format(Style::Upper)), "XII");
+    /// assert_eq!(value.format(Style::Lower).to_string(), "xii"); // `format!("{}")` and `.to_string()` are the same thing
+    /// ```
```

Comment:
Thanks for taking the time to add so much documentation. I can't remember off the top of my head, but do these examples become part of the test suite?

Reply:
Yes, they are tested with `cargo test`